streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -392,6 +490,16 @@ public synchronized final TopologyBuilder addSource(String name, Deserializer ke
             }
         }
 
+        if (offsetReset != null) {
```

Comment:
This block of code looks like a duplicate of lines 416 - 424 - can they be extracted to a method?